UI/window-basic-main-outputs.cpp

```diff
@@ -263,6 +271,85 @@ bool BasicOutputHandler::VirtualCamActive() const
 	return false;
 }
 
+void BasicOutputHandler::UpdateVirtualCamOutputSource()
+{
+	if (!main->vcamEnabled || !virtualCamView)
+		return;
+
+	OBSSourceAutoRelease source;
+
+	switch (main->vcamConfig.type) {
+	case VCamOutputType::InternalOutput:
+		switch (main->vcamConfig.internal) {
+		case VCamInternalType::Default:
+			source = obs_get_output_source(0);
+			break;
+		case VCamInternalType::Preview:
+			OBSSource s = main->GetCurrentSceneSource();
+			obs_source_get_ref(s);
+			source = s.Get();
+			break;
+		}
+		break;
+	case VCamOutputType::SceneOutput:
+		source = obs_get_source_by_name(main->vcamConfig.scene.c_str());
+		break;
+	case VCamOutputType::SourceOutput:
+		OBSSource s =
```

Comment:
`obs_get_source_by_name()` returns a new reference already. `OBSSourceAutoRelease` should be used, and the `obs_source_release()` call below removed.

Reply:
It was like that in the original code, so this review should have been done in the original PR.

I just mostly moved the code for this part. I am thinking of considering this out of scope of this PR.